src/core/namespace.c

```diff
@@ -2369,6 +2369,25 @@ int setup_namespace(
                 drop_unused_mounts(root, mounts, &n_mounts);
         }
 
+        /* We make the host's os-release available via a symlink, so that we can copy it atomically
+         * and readers will never get a half-written version. It is unlikely as it's not updated
+         * often, but better safe than sorry. */
+        if (host_os_release_stage) {
+                symlinks = strv_copy(exec_dir_symlinks);
+                if (!symlinks) {
+                        r = -ENOMEM;
+                        goto finish;
+                }
+
+                r = strv_extend(&symlinks, "/run/host/.os-release-stage/os-release");
```

Comment:
hmm, this really should be a relative symlink, i.e. just point to `.os-release-stage/os-release` rather than `/run/host/.os-release-stage/os-release`

Reply:
The helper makes it relative automatically so the end result will be, also it barfs if it is not absolute as it creates the parent directory as well (I tried with relative at first)

Reply:
```
root@system:/tmp# systemd-run --quiet -t -p RootDirectory=/tmp/r ls -l /run/host/
total 0
lrwxrwxrwx 1 0 0 28 Aug 14 10:08 os-release -> .os-release-stage/os-release
```

Reply:
can you add a comment about that here?